cdap-ui/app/directives/log-viewer/log-viewer.js

```diff
@@ -492,15 +599,6 @@ function LogViewerController ($scope, LogViewerStore, myLogsApi, LOGVIEWERSTORE_
     return entry;
   };
 
-  $scope.$on('$destroy', function() {
-    if (unsub) {
-      unsub();
-    }
-    if(pollPromise){
-      dataSrc.stopPoll(pollPromise.__pollId__);
-      pollPromise = null;
```

Comment:
why is this removed? if the program still running and you exit the state, when will the poll get stopped?

also I believe not unsubscribing will cause a memory leak